x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -58,6 +61,10 @@ public interface Role {
 
     RemoteIndicesPermission remoteIndices();
 
+    boolean hasWorkflowsRestriction();
+
+    Role forWorkflow(String workflow);
```

Comment:
Nit: can we add `@Nullable` to `String workflow` since it is not obvious that it can be null.

Reply:
Sure, I've also added javadoc.